### PR TITLE
Add tests for watch-only mode

### PR DESCRIPTION
**Pull request by jarys**
I cannot set the HEAD to `watchonly` branch :/

### CODE REVIEW

File: zingo-memo/src/lib.rs [hunk 1]
```diff
@@ -4,16 +4,23 @@ use zcash_address::unified::{Address, Container, Encoding, Receiver};
 use zcash_client_backend::address::UnifiedAddress;
 use zcash_encoding::{CompactSize, Vector};
 
-pub const MEMO_VERSION: usize = 0;
-
+/// A parsed memo. Currently there is only one version of this protocol,
+/// which is a list of UAs. The main use-case for this is to record the
+/// UAs sent from, as the blockchain only records the pool-specific receiver
+/// corresponding to the key we sent with.
```
Comment (zancas) [inline zingo-memo/src/lib.rs:10]: This use of `from` and `to` feels swapped to me here.

File: zingo-memo/src/lib.rs [hunk 2]
```diff
@@ -4,16 +4,23 @@ use zcash_address::unified::{Address, Container, Encoding, Receiver};
 use zcash_client_backend::address::UnifiedAddress;
 use zcash_encoding::{CompactSize, Vector};
 
-pub const MEMO_VERSION: usize = 0;
-
+/// A parsed memo. Currently there is only one version of this protocol,
+/// which is a list of UAs. The main use-case for this is to record the
+/// UAs sent from, as the blockchain only records the pool-specific receiver
```
Comment (zancas) [inline zingo-memo/src/lib.rs:9]: I think this "from" should be "to".

Comment (jarys) [reply zingo-memo/src/lib.rs]: This code does not belong to this PR. I accidentally merged `dev` instead of rebasing to it, so the code emerged here.

**Comment by jarys** [comment]
Since #241 haven't been merged yet, I'm pushing tests there.

**Pull request closed without merging**